src/math/matrix.lisp

```diff
@@ -109,80 +109,57 @@
          ,v))
 
 
-(defmethod add ((matrix1 float2x2) (matrix2 float2x2))
-  (make-float2x2
-   (map 'vector #'+
-        (in-vec matrix1)
-        (in-vec matrix2))))
+(defmacro define-matrix-op (name result-type map-op)
+  `(defmethod ,name ((matrix1 ,result-type) (matrix2 ,result-type))
+     (make-instance ',result-type :in-vec
+                    (map 'vector ,map-op
+                         (in-vec matrix1)
+                         (in-vec matrix2)))))
 
-(defmethod add ((matrix1 float3x3) (matrix2 float3x3))
-  (make-float3x3
-   (map 'vector #'+
-        (in-vec matrix1)
-        (in-vec matrix2))))
+(defmacro define-matrix-num-op (name matrix-type map-fun)
+  `(defmethod ,name ((matrix ,matrix-type) (value real))
+     (make-instance ',matrix-type :in-vec
+                    (map 'vector (ax:rcurry ,map-fun value)
+                         (in-vec matrix)))))
 
-(defmethod add ((matrix1 float4x4) (matrix2 float4x4))
-  (make-float4x4
-   (map 'vector #'+
-        (in-vec matrix1)
-        (in-vec matrix2))))
+(defmacro define-matrix-num-revord-op (name matrix-type map-fun)
+  `(defmethod ,name ((value real) (matrix ,matrix-type))
```

Comment:
If you have reword counterparts for all binary operations, it may worth combining both macros together, and define two methods in one macro.
Also, if you have operations which need only one variant, you can pass optional argument to defmacro itself, and make a decision based on its value.